### PR TITLE
Monitoring: add VPCSC system test.

### PR DESCRIPTION
**Pull request by steinwaywhw**
This new file is added in order to test client lib compatibility of VPC SC. The tests are auto-generated from the service definition of the monitoring API. The tests can be run inside or outside of VPC service perimeter. The input to the script should be the following environment variables. 

1. PROJECT_ID: a project that is inside the VPC perimeter. 
2. GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT: a project that is outside the VPC perimeter.

### CODE REVIEW

File: monitoring/tests/system/test_vpcsc.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            list(responses)
+        except exceptions.PermissionDenied as e:
+            return e.message == "Request is prohibited by organization's policy"
+        except:
+            pass
+        return False
+
+    @staticmethod
+    def _do_test(delayed_inside, delayed_outside):
+        if ('GOOGLE_CLOUD_TESTS_IN_VPCSC' in os.environ) and (os.environ['GOOGLE_CLOUD_TESTS_IN_VPCSC'].lower() == 'True'.lower()):
```
Comment (bmccutchon) [inline monitoring/tests/system/test_vpcsc.py]: You can just write `os.environ.get('GOOGLE_CLOUD_TESTS_IN_VPCSC', 'false').lower() == 'true'`.

**Comment by steinwaywhw** [comment]
Hi @tseaver, the VPC team has their own Kokoro workflow to test this. I believe @bmccutchon knows more. Otherwise, if you'd like to run this test in the pull request workflow, please supply these environment variables: `PROJECT_ID=secure-gcp-test-project-4` and `GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT=some GCP test project that you know of`

**Comment by tseaver** [comment]
@tseaver It doesn't seem to me to make sense to have the tests maintained in a separate repository from the location where the CI runs.  @busunkim96, @crwilcox please chime in.

**Comment by steinwaywhw** [comment]
This is a requirement from the VPC team. Many teams are going to implement such tests. I have merged a similar one here: https://github.com/googleapis/google-cloud-java/pull/5006, and the Spanner team has done it here https://github.com/googleapis/google-cloud-java/blob/master/google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITVPCNegativeTest.java.

**Comment by bmccutchon** [comment]
@tseaver Echoing what @steinwaywhw said, we expect the client library teams to write and maintain these tests. The easiest way to do that is to keep them in this repo. I agree that it would make sense to have the CI workflow also be in this repo or run against affected pull requests, but that isn't feasible quite yet.

**Comment by busunkim96** [comment]
@bmccutchon Would it make sense for us to run these tests against PRs in this repo, or is there some limitation that would keep us from doing that? Are there requirements beyond setting the environment variables @steinwaywhw mentioned?

We would probably edit the nox session [here](https://github.com/googleapis/google-cloud-python/blob/f5aec2d2fbb72d613abf5f2c99bdefc501b077d6/monitoring/noxfile.py#L98) to set the required environment variables.

**Comment by steinwaywhw** [comment]
Hi @busunkim96, there are essentially two ways to run these tests: 1) from inside VPCSC, 2) from outside VPCSC. For case 1, you need to setup VPCSC environments, which is non-trivial. The VPC team has a setup, and they can run it. For case 2, you do not need a VPCSC setup, and the tests can be run by simply setting those two environment variables, and the test case still provides valuable informations on the effectiveness of VPC service control.

**Comment by steinwaywhw** [comment]
Friendly ping 👋

**Comment by busunkim96** [comment]
Merging this now. I will create a follow-up PR to set the environment variables to make the tests run.

**Comment by steinwaywhw** [comment]
@busunkim96 Thank you very much! If you have further questions, feel free to reach out to me.

**Pull request merged**